common/txmgr/broadcaster.go

```diff
@@ -600,6 +615,35 @@ func (eb *Broadcaster[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]) hand
 	}
 }
 
+func (eb *Broadcaster[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]) validateOnChainSequence(ctx context.Context, lgr logger.SugaredLogger, errType client.SendTxReturnCode, err error, etx txmgrtypes.Tx[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE], retryCount int) (client.SendTxReturnCode, error) {
+	// Only check if sequence was incremented if broadcast was successful, otherwise return the existing err type
+	if errType != client.Successful {
+		return errType, err
+	}
+	// Transaction sequence cannot be nil here since a sequence is required to broadcast
+	txSeq := *etx.Sequence
+	// Retrieve the latest mined sequence from on-chain
+	nextSeqOnChain, err := eb.client.SequenceAt(ctx, etx.FromAddress, nil)
+	if err != nil {
+		return errType, err
+	}
+	// Check that the transaction count has incremented on-chain to include the broadcasted transaction
+	// Insufficient transaction fee is a common scenario in which the sequence is not incremented by the chain
+	// If the sequence failed to increment and hasn't reached the max retries, return the Underpriced error to try again with a bumped attempt
+	if nextSeqOnChain.Int64() == txSeq.Int64() && retryCount < maxHederaBroadcastRetries {
```

Comment:
nit: in theory if nextSeqOnChain is < txSeq we can mark tx as `Successful`

Reply:
`nextSeqOnChain < txSeq` I think would be an indicator of a nonce gap which wouldn't return a successful `errType` so we would have exited early. But for a belts and braces approach, I added the condition just in case and mark the tx as Fatal in this case.

Reply:
Maybe @dhaidashenko you meant `nextSeqOnChain > txSeq`? 

I agree that `nextSeqOnChain < txSeq` would be incorrect, as it would be an reorg in this case ? otherwise not sure about how this is possible.

Reply:
Ah if that's the case then classifying that as `Successful` is ok because we process `TransactionAlreadyKnown` as the same.
Also unsure if `nextSeqOnChain < txSeq` is a scenario we could realistically hit for this chain. I believe re-orgs aren't possible because of instant finality. But there could be some obscure scenario so this check should help.

Reply:
`Retryable` is also another option for the `nextSeqOnChain < txSeq` case. There are two scenarios:
- We have a nonce gap: fatal will alert us sooner, but we might miss reorg cases
- There was a reorg or RPC was lagging: retryable will handle this, although I found out that Hedera doesn't handle nonce too high errors gracefully, so it will throw a bunch of errors during this process.